utils/stall_free.hh

```diff
@@ -97,5 +97,22 @@ seastar::future<> clear_nested_gently(std::unordered_map<K, Nested>& m) noexcept
     }
 }
 
+template <typename Nested, typename Vector = std::vector<Nested>>
+requires requires (Vector v, typename Vector::reverse_iterator it) {
+    { v.erase(it) } -> std::same_as<typename Vector::reverse_iterator>;
+}
+seastar::future<> clear_nested_gently(Vector& v) noexcept {
+    for (auto it = v.rbegin(); it != v.rend(); it = v.erase(it)) {
+        co_await clear_gently(*it);
+    }
+}
+
+template <typename Container>
+seastar::future<> clear_nested_gently(Container& c) noexcept {
+    for (auto it = c.begin(); it != c.end(); it = c.erase(it)) {
+        co_await clear_gently(*it);
+    }
+}
+
```

Comment:
Can we split the stall_free.hh to a separate patch? It is better to do backport.

Reply:
Backporting into versions that do not have coroutines will require rewriting this without coroutines anyhow.
I don't see enough value in doing this in advance master in this case.

Reply:
done in v3